### PR TITLE
add the test configuration file

### PR DESCRIPTION
**Pull request by sangyx**
Add the test configuration file to generate the test report.

### CODE REVIEW

File: src/.coveragerc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[run]
+branch = True
+source =
+    udiff
```
Comment (hameerabbasi) [inline src/.coveragerc]:
```suggestion
    src/udiff
```

File: src/.coveragerc [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+[report]
+omit =
+    **/tests/
```
Comment (hameerabbasi) [inline src/.coveragerc]:
```suggestion
    **/tests/

```

File: src/pytest.ini [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+addopts = --doctest-modules --junitxml=junit/test-results.xml --cov-report=xml --cov-report=term --cov --cov-report html --cov . --cov-config .coveragerc
+doctest_optionflags= IGNORE_EXCEPTION_DETAIL
```
Comment (hameerabbasi) [inline src/pytest.ini]:
```suggestion
doctest_optionflags= IGNORE_EXCEPTION_DETAIL

```

**Review by hameerabbasi — COMMENTED** [review]
A couple of minor suggestions.

File: src/pytest.ini (hunk 3 above)
Comment (sangyx) [reply src/pytest.ini]: Do you mean I should delete these two lines?
```
**/tests/
doctest_optionflags= IGNORE_EXCEPTION_DETAIL
```
If we change these lines, `pytest.ini` and `.coveragerc` should move to the top-level directory.

Comment (hameerabbasi) [reply src/pytest.ini]: No, I mean there should be a newline at the end of each file. 😄

Comment (hameerabbasi) [reply src/pytest.ini]: And yes, I believe they should be at the top-level.

**Comment by hameerabbasi** [comment]
Thanks, @sangyx.

**Pull request merged**